### DISCUSSION TITLE
Support for allowFreeInput in DateRangePicker

### DISCUSSION THREAD

**Discussion by GermainBergeron** [mantinedev/mantine#2304] [Feature requests]
I'm currently working on integrating Mantine in one project and I'm using the `DateRangePicker`. I got the request to allow users to edit the date in the input but the component doesn't allow that. I have a workaround using two different DatePickers but I'm losing the functionality of viewing the range in the calendar.

When I add the `allowFreeInput` prop there is a type error (I'm using TypeScript), and when I type it doesn't update but the button is is correctly transformed to an Input.

Since it's supported by the `DatePicker` could we add the functionality?

**Comment by rtivital** [comment]
It is not possible to implement it with current components structure without introducing breaking changes. It is a part of 6.x roadmap, so do not expect this feature soon.
